webauthn-rs/src/lib.rs

```diff
@@ -493,6 +511,61 @@ impl Webauthn {
             .map(|(ccr, rs)| (ccr, PasskeyRegistration { rs }))
     }
 
+    /// Initiate the registration of a synced google cloud key on an Android Device.
+    ///
+    /// This function is required as Android's support for Webauthn/Passkeys is broken
+    /// and does not correctly perform authenticator selection for the user. Instead
+    /// of Android correctly presenting the choice to users to select between a
+    /// security key and a synced google cloud key, Android expects the Relying Party
+    /// to pre-select this and send a correct set of options for either security key
+    /// *or* a synced google cloud key.
+    ///
+    /// If you choose to use this function you *MUST* ensure that the device you are
+    /// contacting is an Android device, and you *MUST* provide the user the choice
+    /// on your site ahead of time to choose between a security key / screen unlock
+    /// (triggered by [`start_passkey_registration`](Webauthn::start_passkey_registration))
+    /// or a synced google cloud key (triggered by this function).
+    #[cfg(feature = "workaround-android-specific-issues")]
+    pub fn start_android_only_passkey_registration(
```

Comment:
This should be named something like "Google Passkey" rather than just "Android", because it is not "make passkey work on Android" but "allow Google Passkey stored in Google Password Manager".

This is also specific to Android devices with GMS (which covers the majority of devices, generally excluding those sold in PRC), just to make things more fun.